sources/api/prairiedog/src/bootconfig.rs

```diff
@@ -0,0 +1,410 @@
+        Some(boot_settings) => {
+            info!("Generating initrd boot config from boot settings");
+            trace!("Boot settings: {:?}", boot_settings);
+            let bootconfig = serialize_boot_settings_to_boot_config(&boot_settings)?;
+            trace!("Serializing boot config string: {}", bootconfig);
+            bootconfig.into_bytes()
+        }
+        None => {
+            // If we don't have any boot settings, write out an initrd with default boot config contents
+            trace!("Serializing boot config string: {}", DEFAULT_BOOTCONFIG_STR);
+            DEFAULT_BOOTCONFIG_STR.to_string().into_bytes()
+        }
+    };
+    let initrd = generate_initrd(&bootconfig_bytes)?;
+    trace!("Writing initrd image file: {:?}", initrd);
+    tokio::fs::write(BOOTCONFIG_INITRD_PATH, &initrd)
+        .await
+        .context(error::WriteInitrdSnafu)?;
+    Ok(())
+}
+
+/// Retrieves boot config related Bottlerocket settings. If they don't exist in the settings model,
+/// we return `None` instead.
+async fn get_boot_config_settings<P>(socket_path: P) -> Result<Option<BootSettings>>
+where
+    P: AsRef<Path>,
+{
+    let uri = "/settings";
+    let settings: serde_json::Value =
+        schnauzer::get_json(socket_path, uri, Some(("prefix", "boot")))
+            .await
+            .context(error::RetrieveSettingsSnafu)?;
+
+    match settings.get("boot") {
+        None => Ok(None),
+        Some(boot_settings_val) => Ok(Some(
+            serde_json::from_value(boot_settings_val.to_owned())
+                .context(error::BootSettingsFromJsonValueSnafu)?,
+        )),
+    }
+}
+
+/// Reads `/proc/bootconfig` and populates the Bottlerocket boot settings based on the existing boot config data
+pub(crate) async fn generate_boot_settings() -> Result<()> {
+    let proc_bootconfig = match tokio::fs::read_to_string(PROC_BOOTCONFIG).await {
+        Ok(s) => s,
+        Err(e) => {
+            // If there's no `/proc/bootconfig`, then the user hasn't provisioned any kernel boot configuration.
+            if e.kind() == io::ErrorKind::NotFound {
+                // No work to be done
+                return Ok(());
+            }
+            return Err(e).context(error::ReadFileSnafu {
+                path: PROC_BOOTCONFIG,
+            });
+        }
+    };
+    debug!(
+        "Generating kernel boot config settings from `{}`",
+        PROC_BOOTCONFIG
+    );
+    println!("{}", boot_config_to_boot_settings_json(&proc_bootconfig)?);
+
+    Ok(())
+}
+
+/// Parses out a valid boot config value
+fn parse_value(input: &str) -> Result<BootConfigValue> {
+    let input = input.trim();
+    let quoted = (input.starts_with('"') && input.ends_with('"'))
+        || (input.starts_with('\'') && input.ends_with('\''));
+    let chars_that_require_quotes = ['\'', '"', '\n', ',', ';', '#', '}'];
+    let valid_value = input
+        .chars()
+        .all(|c| c.is_ascii() && (quoted || !chars_that_require_quotes.contains(&c)));
```

Comment:
It might be better to get rid of these checks entirely.

We can remove the outer quotes (if present) and then delegate the work to the `try_into()` call at the end, which will repeat these tests.

That would also let us remove the `InvalidBootConfigValueSnafu` error here which has different criteria than the modeled type.

Reply:
I think we need these checks to guard against unquoted values with disallowed delimiters. 
>Each value only contains printable characters or spaces except for delimiters such as semi-colon (;), new-line (\n), comma (,), hash (#) and closing brace (}).

The modeled type does not check for that, it assumes all `BootConfigValues` are eventually double/single quoted so it only ensures the value is printable ASCII sans control chars.

For example, if we're parsing a boot config file:
```
kernel = #bang
```

Without this check, it'll be allowed.

Reply:
Ah, gotcha. Even though we could accept that value - since we know we'll end up quoting it - it's not a valid input so we should reject it.